junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -139,9 +139,10 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        prepared.countDown();
-        startable.countDown();
-        finished.countDown();
+        // Commented because of issue https://github.com/k3po/k3po/issues/391
```

Comment:
The situation I encountered was the following: the prepare failed, the main thread was in awaitPrepared, the script thread calls notifyException. First it calls prepare.countDown() so the main thread gets unlocked and fails properly. But the script thread then calls testThread.interrupt() and the interrupt status is set for the thread. Because the main thread is not anymore in an await, nobody is clearing the interrupt status. 
The test ends with the expected result (failed in prepare), next test starts. When the first wait is encountered, the InterruptedException is thrown, although nothing caused it on the new test. So the new test fails with some strange exception (some InterruptException with no cause) and a NullPointerException is shown in the log because there is no connection. At least this is what happened in my tests. I guess that different exceptions/errors can occur.
So, in order to let the InterruptedException to occur, we need to skip unlocking the latches.